lib/scopes/form526_submission_state.rb

```diff
@@ -0,0 +1,106 @@
+      scope :in_process, lambda {
+        where(submitted_claim_id: nil)
+          .where(backup_submitted_claim_id: nil)
+          .where(arel_table[:created_at].gt(Form526Submission::MAX_PENDING_TIME.ago))
+          .where.not(id: remediated.pluck(:id))
+          .where.not(id: with_exhausted_backup_jobs.pluck(:id))
+      }
+
+      scope :accepted_to_primary_path, lambda {
+        where.not(submitted_claim_id: nil)
+      }
+      scope :accepted_to_backup_path, lambda {
+        where.not(backup_submitted_claim_id: nil)
+             .where(
+               backup_submitted_claim_status: [
+                 backup_submitted_claim_statuses[:accepted],
+                 backup_submitted_claim_statuses[:paranoid_success]
+               ]
+             )
+      }
+      scope :rejected_from_backup_path, lambda {
+        where.not(backup_submitted_claim_id: nil)
+             .where(backup_submitted_claim_status: backup_submitted_claim_statuses[:rejected])
+      }
+
+      scope :remediated, lambda {
+        ids = joins(
+          "INNER JOIN (
+            SELECT form526_submission_id, MAX(created_at) AS max_created_at
+            FROM form526_submission_remediations
+            GROUP BY form526_submission_id
+          ) AS latest_remediations
+          ON form526_submissions.id = latest_remediations.form526_submission_id"
+        ).joins(
+          "INNER JOIN form526_submission_remediations
+          ON form526_submission_remediations.form526_submission_id = latest_remediations.form526_submission_id
+          AND form526_submission_remediations.created_at = latest_remediations.max_created_at"
+        ).where(
+          form526_submission_remediations: { success: true }
+        ).select(:id)
+        where(id: ids) # HACK: allow clean scope joining. Could be removed in favor of Arel
+      }
+
+      scope :with_exhausted_primary_jobs, lambda {
+        joins(:form526_job_statuses)
+          .where(submitted_claim_id: nil)
+          .where(form526_job_statuses: { job_class: 'SubmitForm526AllClaim' })
+          .where(form526_job_statuses: { status: Form526JobStatus::FAILURE_STATUSES.values })
+      }
+      scope :with_exhausted_backup_jobs, lambda {
+        joins(:form526_job_statuses)
+          .where(backup_submitted_claim_id: nil)
+          .where(form526_job_statuses: { job_class: 'BackupSubmission' })
+          .where(form526_job_statuses: { status: Form526JobStatus::FAILURE_STATUSES.values })
+      }
+
+      # Documentation describing the purpose of 'paranoid success' and 'success by age'
+      # https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/engineering_research/paranoid_success_submissions.md
+      scope :paranoid_success_type, lambda {
+        where.not(backup_submitted_claim_id: nil)
+             .where(backup_submitted_claim_status: backup_submitted_claim_statuses[:paranoid_success])
+             .where.not(id: success_by_age.pluck(:id))
```

Comment:
Do you want to reference the `success_by_age` scope in this scope or, instead, limit results to those submissions that were created less than a year ago?

Reply:
so it's really the same effect... My thinking is that *logically*, this scope is "everying that is is 'paranoid success' that has not yet become successful by age." therefore, it's reasonable to say where not success by age.  Also, it couples them, which is good... that way if we change the 'age' to e.g. 2 years, we change it in one place, not two.

Reply:
Makes sense but, to avoid the redundant query conditions, could the 2 scopes be improved by extracting the common parts into a separate scope or using a shared method?

Reply:
I considered that.. but imho it's better to have a bit of redundancy than a scope that should never be used in isolation. my fear is that adding a scope just for a subquery will be confusing, and I'd  wnat to be sure we don't have future generations think `labled_as_paranoid_success` (or whatever it would be called) is more relevant than `paranoid_succes_type`. idk, what do you think? I'm on the fence

Reply:
You make a valid case for your decision to leave as is.